src/pages/sc-cycle-counts/Forms/CycleCountsForm.js

```diff
@@ -0,0 +1,472 @@
+    stack({
+      Component: WorkFlow,
+      props: {
+        functionId: SystemFunction.StockCount,
+        recordId: formik.values.recordId
+      },
+      width: 950,
+      title: 'Workflow'
+    })
+  }
+
+  async function getCycleCounts(recordId) {
+    return await getRequest({
+      extension: SCRepository.StockCount.get,
+      parameters: `_recordId=${recordId}`
+    })
+  }
+
+  async function refetchForm(recordId) {
+    const res2 = await getCycleCounts(recordId)
+    res2.record.date = formatDateFromApi(res2.record.date)
+
+    return res2;
+  }
+
+  const isClosed = formik.values.wip === 2
+  const isPosted = formik.values.status === 3
+
+  const onClose = async recId => {
+    try {
+      const res = await postRequest({
+        extension: SCRepository.StockCount.close,
+        record: JSON.stringify({ recordId: recId })
+      })
+
+      if (res.recordId) {
+        toast.success(platformLabels.Closed)
+        invalidate()
+        const res2 = await refetchForm(res.recordId)
+        formik.setValues(res2.record)
+        setStore(prevStore => ({
+          ...prevStore,
+          isClosed: res2.record.wip === 2,
+        }))
+      }
+    } catch (error) {}
+  }
+
+  async function onReopen() {
+    try {
+      const res = await postRequest({
+        extension: SCRepository.StockCount.reopen,
+        record: JSON.stringify(formik.values)
+      })
+
+      if (res.recordId) {
+        toast.success(platformLabels.Reopened)
+        invalidate()
+        const res2 = await refetchForm(res.recordId)
+        formik.setValues(res2.record)
+        setStore(prevStore => ({
+          ...prevStore,
+          isClosed: res2.record.wip === 2,
+        }))
+      }
+    } catch (error) {}
+  }
+
+  const onPost = async () => {
+    try {
+      const copy = { ...formik.values }
+      copy.date = formatDateToApi(copy.date)
+
+      const res = await postRequest({
+        extension: SCRepository.StockCount.post,
+        record: JSON.stringify(copy)
+      })
+
+      toast.success(platformLabels.Posted)
+      invalidate()
+
+      const res2 = await refetchForm(res.recordId)
+      formik.setValues(res2.record)
+      setStore(prevStore => ({
+        ...prevStore,
+        isPosted: res2.record.status === 3,
+      }))
+      window.close()
+    } catch (error) {}
+  }
+
+  const actions = [
+    {
+      key: 'WorkFlow',
+      condition: true,
+      onClick: onWorkFlowClick,
+      disabled: !editMode
+    },
+    {
+      key: 'Close',
+      condition: !isClosed,
+      onClick: () => {
+        onClose(formik.values.recordId)
+      },
+      disabled: isClosed || !editMode || isPosted
+    },
+    {
+      key: 'Reopen',
+      condition: isClosed,
+      onClick: onReopen,
+      disabled: !isClosed || !editMode
+    },
+    {
+      key: 'Post',
+      condition: true,
+      onClick: onPost,
+      disabled: isPosted || !editMode || !isClosed
+    }
+  ]
+
+  async function getData(recordId) {
```

Comment:
you have duplicate functions that behave the save getCycleCounts & getData, remove one of them